source/DomainTheory/Examples/Ordinals.lagda

```diff
@@ -0,0 +1,172 @@
+(1) Every ordinal α is equal to the supremum of the successors of its initial
+    segments, i.e. α = sup (λ a → (α ↓ a) +ₒ 𝟙ₒ)
+(2) Every successor ordinal, i.e. every ordinal of the form β +ₒ 𝟙ₒ is compact.
+(3) The family in (1) can be "directified" by taking finite joins which
+    preserves compactness.
+
+\begin{code}
+
+successor-ordinals-are-compact : (α : Ordinal 𝓤)
+                               → is-compact Ordinals-DCPO (α +ₒ 𝟙ₒ)
+successor-ordinals-are-compact α I β δ l =
+ ∥∥-functor (λ (i , b , eq₂) → ⦅3⦆ (i , b , (eq₁ ∙ eq₂))) ⦅2⦆
+  where
+   ⦅1⦆ : Σ s ꞉ ⟨ sup β ⟩ , α ＝ sup β ↓ s
+   ⦅1⦆ = ⊴-gives-≼ (α +ₒ 𝟙ₒ) (sup β) l α (successor-increasing α)
+   s : ⟨ sup β ⟩
+   s = pr₁ ⦅1⦆
+   eq₁ : α ＝ sup β ↓ s
+   eq₁ = pr₂ ⦅1⦆
+   ⦅2⦆ : ∥ Σ i ꞉ I , Σ b ꞉ ⟨ β i ⟩ , sup β ↓ s ＝ β i ↓ b ∥
+   ⦅2⦆ = initial-segment-of-sup-is-initial-segment-of-some-component β s
+   ⦅3⦆ : (Σ i ꞉ I , Σ b ꞉ ⟨ β i ⟩ , α ＝ β i ↓ b)
+       → Σ i ꞉ I , (α +ₒ 𝟙ₒ) ⊴ β i
+   ⦅3⦆ (i , b , refl) = i , upper-bound-of-successors-of-initial-segments (β i) b
+
+private
+ module _ (α : Ordinal 𝓤) where
+  family : ⟨ α ⟩ → Ordinal 𝓤
+  family a = (α ↓ a) +ₒ 𝟙ₒ
+
+Ordinals-DCPO-is-algebraic' : structurally-algebraic Ordinals-DCPO
+Ordinals-DCPO-is-algebraic' =
+ record
+  { index-of-compact-family = λ α → List ⟨ α ⟩
+  ; compact-family = λ α → directify (family α)
+  ; compact-family-is-directed = λ α → directify-is-directed (family α)
+  ; compact-family-is-compact = λ α → directify-is-compact
+                                       (family α)
+                                       (λ a → successor-ordinals-are-compact (α ↓ a))
+  ; compact-family-∐-＝ = eq
+  }
+   where
+    eq : (α : Ordinal 𝓤)
+       → ∐ Ordinals-DCPO (directify-is-directed (family α)) ＝ α
+    eq α = ∐ Ordinals-DCPO (directify-is-directed (family α)) ＝⟨ I ⟩
+           sup (family α)                                     ＝⟨ II ⟩
+           α                                                  ∎
+     where
+      II = (supremum-of-successors-of-initial-segments pt sr α) ⁻¹
+      I = sups-are-unique _⊴_
+           (pr₁ (axioms-of-dcpo Ordinals-DCPO)) (family α)
+           (directify-sup' (family α)
+             (∐ Ordinals-DCPO δ) (∐-is-sup Ordinals-DCPO δ))
+           (sup-is-least-upper-bound (family α))
+       where
+        δ : is-Directed Ordinals-DCPO (directify (family α))
+        δ = directify-is-directed (family α)
+
+Ordinals-DCPO-is-algebraic : is-algebraic-dcpo Ordinals-DCPO
+Ordinals-DCPO-is-algebraic = ∣ Ordinals-DCPO-is-algebraic' ∣
+
+\end{code}
+
+Unlike many other examples, such as the dpcos in the Scott model of PCF, or
+Scott's D∞, the dpco of ordinals, while algebraic, does not have a small
+(compact) basis. If it did, we could take the join of all the basis elements to
+obtain a greatest ordinal, which does not exist.
+
+\begin{code}
+
+Ordinals-DCPO-has-no-small-basis : ¬ (has-unspecified-small-basis Ordinals-DCPO)
```

Comment:
Nice to have this counterexample formalized. I have yet to find a frame that does not have a small basis.

Reply:
> Nice to have this counterexample formalized. I have yet to find a frame that does not have a small basis.

What about not-not stable propositions (without propositional resizing)?

Reply:
The type of not-not-stable propositions would not work since the Booleans form a basis for them, as they do for $\Omega$. However, there is a resizing taboo in this direction, which might be what you had in mind @UlrikBuchholtz.

In [our preprint](https://arxiv.org/abs/2402.03134) on the patch topology with @martinescardo and @IgorArrieta, we have two equivalent definitions of the notion of spectral locale:

1. The one in Definition 4.5 which says that the type of compact opens forms a small basis closed under binary meets.
2. The one that says there merely exists a small spectral basis (defined in Definition 4.9) for the locale in consideration.

In Theorem 4.17, it is shown that they coincide, but this crucially relies on the locale in consideration being compact.

For zero-dimensional locales (not necessarily Stone), one can consider the analogues of these definitions and ask if they coincide.

They don’t coincide and a counterexample to this is $\Omega_{\neg\neg}$; its type of clopens is not small but there exists a small basis consisting of clopens. By going from $\Omega$ to $\Omega_{\neg\neg}$ (the smallest dense sublocale of $\Omega$), the size of the clopens increases from just $2$ to $\Omega$-many because everything in $\Omega_{\neg\neg}$ is a clopen. So if the two definitions coincide on it, then $\Omega_{\neg\neg}$-resizing holds.

This counterexample was found by @IgorArrieta when I asked him I was looking for such a counterexample.

Reply:
Interesting. But is there an example where provably there isn't a small base?

Reply:
@martinescardo No, there is no example where the nonexistence of a small base is provable. There is not even an example where the existence of a small basis gives a resizing taboo.